File: games/guess_number.py
import random
from flask import Blueprint, render_template, request, jsonify, session

guess_number_bp = Blueprint('guess_number', __name__)

@guess_number_bp.route('/')
def guess_number():
    session['secret_number'] = random.randint(1, 100)
    session['attempts'] = 0
    return render_template('guess_number.html')

@guess_number_bp.route('/play', methods=['POST'])
def play():
    user_guess = int(request.form['guess'])
    secret_number = session.get('secret_number')
    session['attempts'] += 1

    if user_guess < secret_number:
        message = "Бага байна! Дээшлүүлнэ үү."
        status = "low"
    elif user_guess > secret_number:
        message = "Их байна! Доошлуулна уу."
        status = "high"
    else:
        message = f"Баяр хүргэе! Та {session['attempts']} оролдлогоор зөв таалаа!"
        status = "correct"
        session.pop('secret_number', None)

    return jsonify({"message": message, "status": status})

File: static/js/guess_number.js
document.addEventListener('DOMContentLoaded', () => {
    const guessForm = document.getElementById('guessForm');
    const userGuessInput = document.getElementById('userGuess');
    const resultParagraph = document.getElementById('result');
    const resetGameButton = document.getElementById('resetGame');

    if (guessForm) {
        guessForm.addEventListener('submit', async (e) => {
            e.preventDefault();
            const guess = userGuessInput.value;

            const response = await fetch('/guess_number/play', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded',
                },
                body: `guess=${guess}`
            });

            const data = await response.json();
            resultParagraph.textContent = data.message;
            resultParagraph.style.color = '#333'; // Default color

            if (data.status === "correct") {
                resultParagraph.style.color = 'green';
                guessForm.style.display = 'none'; // Нуух
                resetGameButton.style.display = 'block'; // Харуулах
            } else if (data.status === "low" || data.status === "high") {
                resultParagraph.style.color = 'orange';
            }
        });
    }

    if (resetGameButton) {
        resetGameButton.addEventListener('click', () => {
            window.location.reload(); // Хуудсыг дахин ачаалах
        });
    }
});